### ISSUE TITLE
Incorrect parsing

### ISSUE THREAD

**Issue by byteZorvin** [gaetbout/starknet-mutation-testing#1]
- It incorrectly assumes type angular brackets as `<` operator causing a build failure
- This happens in component
![Screenshot 2024-08-12 at 6 42 27 PM](https://github.com/user-attachments/assets/e781e3f7-b0d0-424a-9363-4227a6ce145f)

**Comment by gaetbout** [comment]
Yes, this is normal at the moment. 
I just do a basic (dumb) replace and some replacement can lead to build failures. This is a small shortcut I took.
You should monitor those build failure but I believe that all of them can be safely ignored. 
For example if you have such piece of code (code taken from Alexandria):
```rust
impl BigIntegerToAsciiTraitImpl<
    T,
    +PartialOrd<T>,
    +DivRem<T>,
    +Into<T, felt252>,
    ...
> of ToAsciiTrait<T, Array<felt252>> {

```
It'll replace it to: 

```rust
impl BigIntegerToAsciiTraitImpl<
    T,
    +PartialOrd<T>,
    +DivRem<T>,
    +Into<T, felt252>,
    ...
>= of ToAsciiTrait<T, Array<felt252>> {

```
Which will lead to a build failure

**Comment by byteZorvin** [comment]
Makes sense, do you plan to avoid such failures in future roadmap. Is there an alternate way or this dump replace is faster/efficient than other ways(not sure which, may be some kind of grammar parsing)

**Comment by gaetbout** [comment]
I didn't look much into yet, I was mostly trying to see whether this tool was used and useful first.
If it is, I'll def improve that part

**Comment by byteZorvin** [comment]
The project seem quite useful, let me know if I can contribute in any way. Thanks in advance
